tests/testthat/test-reorder.R

```diff
@@ -57,3 +57,52 @@ test_that("fct_inseq gives error for non-numeric levels", {
   f <- factor(c("c", "a", "a", "b"))
   expect_error(levels(fct_inseq(f)), "level must be coercible to numeric")
 })
+
+test_that("fct_reordern works for all scenarios (fix #16)", {
+  A <- c(3, 3, 2, 1)
+  B <- c("A", "B", "C", "D")
+  f <- c("A", "B", "C", "D")
+  f_factor_ordered <- factor(f, ordered = TRUE)
+  expect_equal(
+    fct_reordern(.f = f, A, B),
+    factor(f, levels = c("D", "C", "A", "B"))
+  )
+  expect_equal(
```

Comment:
You can remove the tests for `decreasing = TRUE` because you are just testing the behaviour of an existing function.

Reply:
Removed except for one to confirm that interaction between the two works as decreasing sorting is an important behavior.

Reply:
That adds a dependency on dplyr — instead you'll need to implement a local version of `desc()` inside the test, e.g. `desc <- function(x) -xtfrm(x)`

Reply:
For my education, I thought that since the dplyr call is in a test and dplyr is already in the DESCRIPTION Suggests field, I thought that this was OK.  (Similarly, since I have it in the examples in the help, I thought that the same applied there.)  I'm happy to implement that way, but my hope was that this would be simpler for the user (via help) and future maintainer (via tests) to use.

Whichever way you confirm, I'll update to that.